File: android-cuantrack/app/src/main/java/com/rork/cuantrackandroid/models/Account.kt
package com.rork.cuantrackandroid.models

import java.util.UUID

data class Account(
    val id: UUID = UUID.randomUUID(),
    val bankName: String = "",
    val accountHolder: String = "",
    val balance: Double = 0.0,
    val accountNumber: String = ""
) {
    val formattedBalance: String
        get() {
            val nf = java.text.NumberFormat.getNumberInstance(java.util.Locale("id", "ID"))
            nf.maximumFractionDigits = 0
            return "Rp ${nf.format(balance)}"
        }
}

File: ios-cuantrack/CuanTrack/Models/Account.swift
//
//  Account.swift
//  CuanTrack
//

import Foundation

struct Account: Identifiable, Codable {
    let id: UUID
    var bankName: String
    var accountHolder: String
    var balance: Double
    var accountNumber: String

    init(
        id: UUID = UUID(),
        bankName: String = "",
        accountHolder: String = "",
        balance: Double = 0,
        accountNumber: String = ""
    ) {
        self.id = id
        self.bankName = bankName
        self.accountHolder = accountHolder
        self.balance = balance
        self.accountNumber = accountNumber
    }

    var formattedBalance: String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        formatter.groupingSeparator = "."
        formatter.decimalSeparator = ","
        formatter.minimumFractionDigits = 0
        formatter.maximumFractionDigits = 0
        return "Rp \(formatter.string(from: NSNumber(value: balance)) ?? "0")"
    }
}
